### PR TITLE
planner: add more test cases for join hint compatibility

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/36600

Problem Summary:
add more test cases for join hint compatibility

### What is changed and how it works?
add more test cases for join hint compatibility

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/casetest/testdata/integration_suite_out.json [hunk 1]
```diff
@@ -6711,6 +6711,954 @@
...
+          "    │   ├─IndexFullScan(Build) 10000.00 cop[tikv] table:t, index:idx_a(a) keep order:true, stats:pseudo",
+          "    │   └─TableRowIDScan(Probe) 10000.00 cop[tikv] table:t keep order:false, stats:pseudo",
+          "    └─Projection(Probe) 9990.00 root  test.t1.a, test.t1.b",
+          "      └─IndexLookUp 9990.00 root  ",
+          "        ├─IndexFullScan(Build) 10000.00 cop[tikv] table:t1, index:idx_a(a) keep order:true, stats:pseudo",
+          "        └─Selection(Probe) 9990.00 cop[tikv]  not(isnull(test.t1.b))",
+          "          └─TableRowIDScan 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo"
+        ],
+        "Warn": [
+          "[planner:1815]There are no matching table names for (t2) in optimizer hint /*+ LEADING(t2) */. Maybe you can use the table alias name"
```
Comment (Reminiscent) [inline planner/core/casetest/testdata/integration_suite_out.json:7411]: Need further check here.

Comment (Reminiscent) [reply planner/core/casetest/testdata/integration_suite_out.json:7411]: This is another issue. I'll record and fix it in a separate PR.

File: planner/core/casetest/testdata/integration_suite_out.json [hunk 2]
```diff
@@ -6711,6 +6711,881 @@
...
+          "  │     └─TableRowIDScan(Probe) 10000.00 cop[tikv] table:t keep order:false, stats:pseudo",
+          "  └─MaxOneRow(Probe) 12500.00 root  ",
+          "    └─StreamAgg 12500.00 root  funcs:sum(Column#25)->Column#10",
+          "      └─Projection 125000.00 root  cast(test.t2.a, decimal(10,0) BINARY)->Column#25",
+          "        └─IndexLookUp 125000.00 root  ",
+          "          ├─IndexRangeScan(Build) 125000.00 cop[tikv] table:t2, index:idx_b(b) range: decided by [eq(test.t2.b, test.t1.b)], keep order:false, stats:pseudo",
+          "          └─TableRowIDScan(Probe) 125000.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ],
+        "Warn": [
+          "[planner:1815]There are no matching table names for (t2) in optimizer hint /*+ LEADING(t2) */. Maybe you can use the table alias name"
```
Comment (Reminiscent) [inline planner/core/casetest/testdata/integration_suite_out.json:7449]: ditto

File: planner/core/casetest/testdata/integration_suite_out.json [hunk 3]
```diff
@@ -6711,6 +6711,881 @@
...
+          "              ├─IndexFullScan_68(Build) 9990.00 cop[tikv] table:t2, index:idx_b(b) keep order:true, stats:pseudo",
+          "              └─Selection_70(Probe) 9980.01 cop[tikv]  not(isnull(test.t2.a))",
+          "                └─TableRowIDScan_69 9990.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ],
+        "Warn": [
+          "[planner:1815]We can only use one leading hint at most, when multiple leading hints are used, all leading hints will be invalid",
+          "[planner:1815]There are no matching table names for (t2) in optimizer hint /*+ MERGE_JOIN(t2, t2) */ or /*+ TIDB_SMJ(t2, t2) */. Maybe you can use the table alias name",
+          "[planner:1815]We can only use one leading hint at most, when multiple leading hints are used, all leading hints will be invalid",
+          "[planner:1815]There are no matching table names for (t2, t2) in optimizer hint /*+ MERGE_JOIN(t2, t2, t2) */ or /*+ TIDB_SMJ(t2, t2, t2) */. Maybe you can use the table alias name"
+        ]
```
Comment (Reminiscent) [inline planner/core/casetest/testdata/integration_suite_out.json:7301]: ditto

**Review by qw4990 — APPROVED** [review]

**Review by fzzf678 — APPROVED** [review]

**Pull request merged**